wait/testdata/main.go

```diff
@@ -19,13 +21,28 @@ func main() {
 	})
 
 	mux.HandleFunc("/ping", func(w http.ResponseWriter, req *http.Request) {
-		data, _ := ioutil.ReadAll(req.Body)
-		if bytes.Equal(data, []byte("ping")) {
-			w.WriteHeader(http.StatusOK)
-			_, _ = w.Write([]byte("pong"))
-		} else {
-			w.WriteHeader(http.StatusBadRequest)
+		auth := req.Header.Get("Authorization")
```

Comment:
How would you see defining different endpoints for both types:

- `/ping`: old behaviour
- `/authenticated-ping`: with Basic Auth

That would decouple the logic (and the future maintenance) of that code. That would mean adding a separate test hitting the new endpoint.

Does it make sense to you?

Reply:
Other than that, I'm OK with this PR 👍

Reply:
Hey @nevins-b sorry for the late response. Did you check my comment about having a separate handler for the authenticated ping? Once there, I think this is ready to go.

Thanks!!

Reply:
Hey @nevins-b I've pushed a commit on top of yours, separating the two handlers. Hope you can 👀